{"metadata": {"language_info": {"name": "python", "version": "3.9.16", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "pygments_lexer": "ipython3", "nbconvert_exporter": "python", "file_extension": ".py"}, "kernelspec": {"name": "python3", "display_name": "Python 3 (ipykernel)", "language": "python"}}, "nbformat_minor": 4, "nbformat": 4, "cells": [{"cell_type": "markdown", "source": "# TCGA Mutation and Clinical Data to RDF Knowledge Graph\n\n**Author:** Chiao-Feng Lin  \n**Email:** clin at dnanexus.com  \n**Date:** Oct 21 2023\n\n## Description\n\nThis Jupyter Notebook is part of the Data Management for Transformer Models hackathon project. It demonstrates the conversion of The Cancer Genome Atlas (TCGA) mutation and clinical sample data into a RDF (Resource Description Framework) knowledge graph. The resulting knowledge graph can be used for semantic representation and querying of TCGA data.\n\n## Project Details\n\n- **Hackathon Name:** Data Management for Transformer Models\n- **Team Name:** Cohort-based vcfs to Knowledge Graphs\n- **Team Members:**\n  - Chiao-Feng Lin (team lead)\n  - Rachit Kumar\n  - Soham Shirolkar \n  - Aniket Naik\n\n## Dependencies\n\nThis notebook relies on the following Python libraries and tools:\n- rdflib, pandas\n- BioPortal API key\n\n## License\n\nThis notebook is provided under the MIT license.", "metadata": {}}, {"cell_type": "code", "source": "### Making BioPortal API calls requires BioPortal API key\n# Please acquire one and replace it here.\napikey=\"BioPortal-API-Key\"", "metadata": {"trusted": true, "tags": []}, "execution_count": 1, "outputs": []}, {"cell_type": "code", "source": "pip install rdflib", "metadata": {"trusted": true, "tags": []}, "execution_count": 2, "outputs": [{"name": "stdout", "text": "\u001b[33mWARNING: The directory '/home/dnanexus/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.\u001b[0m\u001b[33m\n\u001b[0mCollecting rdflib\n  Downloading rdflib-7.0.0-py3-none-any.whl (531 kB)\n\u001b[2K     \u001b[90m\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u001b[0m \u001b[32m531.9/531.9 kB\u001b[0m \u001b[31m55.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hRequirement already satisfied: pyparsing<4,>=2.1.0 in /opt/conda/lib/python3.9/site-packages (from rdflib) (3.1.1)\nCollecting isodate<0.7.0,>=0.6.0\n  Downloading isodate-0.6.1-py2.py3-none-any.whl (41 kB)\n\u001b[2K     \u001b[90m\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u2501\u001b[0m \u001b[32m41.7/41.7 kB\u001b[0m \u001b[31m198.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hRequirement already satisfied: six in /opt/conda/lib/python3.9/site-packages (from isodate<0.7.0,>=0.6.0->rdflib) (1.16.0)\nInstalling collected packages: isodate, rdflib\nSuccessfully installed isodate-0.6.1 rdflib-7.0.0\n\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n\u001b[0mNote: you may need to restart the kernel to use updated packages.\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "import pandas as pd\nfrom rdflib import Graph, Literal, Namespace, RDF, URIRef\nimport requests\n", "metadata": {"trusted": true, "tags": []}, "execution_count": 3, "outputs": []}, {"cell_type": "code", "source": "def query_mondo (cancer_type_query=None):\n    \"\"\" Returns a Mondo ID for a cancer type\"\"\"\n    # BioPortal API endpoint for searching MONDO ontology classes\n    bioportal_api_url = \"https://data.bioontology.org/ontologies/MONDO/classes\"\n\n    # Define the cancer type we want to search for\n    #cancer_type_query = \"Colon Adenocarcinoma\" \n\n    # Prepare the query parameters\n    params = {\n        \"q\": cancer_type_query,\n        \"apikey\": apikey,  # Replace with your BioPortal API key\n    }\n\n    # Send an HTTP GET request to the BioPortal API\n    response = requests.get(bioportal_api_url, params=params)\n\n    # Check if the request was successful (status code 200)\n    if response.status_code == 200:\n        # Parse the JSON response\n        data = response.json()\n\n        # Extract the concept ID for the first matching result (if available)\n        if \"collection\" in data and len(data[\"collection\"]) > 0:\n            first_result = data[\"collection\"][0]\n            concept_id = first_result.get(\"@id\", \"N/A\")\n            #print(f\"Concept ID for '{cancer_type_query}': {concept_id}\")\n            return concept_id\n        else:\n            print(f\"No matching results found for '{cancer_type_query}'\")\n    else:\n        print(f\"Failed to query the BioPortal API. Status code: {response.status_code}\")\n    ", "metadata": {"trusted": true, "tags": []}, "execution_count": 4, "outputs": []}, {"cell_type": "markdown", "source": "## Read TCGA mutation file", "metadata": {}}, {"cell_type": "code", "source": "mut_df = pd.read_csv(\"/mnt/project/data/coad_cptac_2019/data_mutations.txt\",sep=\"\\t\")\nmut_df", "metadata": {"trusted": true, "tags": []}, "execution_count": 5, "outputs": [{"name": "stderr", "text": "/tmp/ipykernel_69/2332604165.py:1: DtypeWarning: Columns (4,37,38,39,43,47,49,52,53,58,59,62,64,71,72,73,74,75,77,80,83,88,92,96,105,116,117,118,132,138,155,168,173,176,179,180,181,182,183,187,188,190,197,198,205,206,207,208,209,210,211,212,213,214,215,216,217,218,219,220,221,222,223,224,225,226,227,228,229,230,231,232,233,234,235,236,237,238,239,240,241,242,243,244,245,246,247,248,249,250,251,252,253,254,255,256,257,258,259,260,261,262,263,264,265,266,267,268,269,270,271,272,273,274,275,276,277,278,279,280,281,282,283,284,285,286) have mixed types. Specify dtype option on import or set low_memory=False.\n  mut_df = pd.read_csv(\"/mnt/project/data/coad_cptac_2019/data_mutations.txt\",sep=\"\\t\")\n", "output_type": "stream"}, {"execution_count": 5, "output_type": "execute_result", "data": {"text/plain": "      Hugo_Symbol  Entrez_Gene_Id  Center NCBI_Build Chromosome  \\\n0          PTPN22           26191     NaN     GRCh37          1   \n1           KIF17           57576     NaN     GRCh37          1   \n2        C1orf167          284498     NaN     GRCh37          1   \n3            CD1D             912     NaN     GRCh37          1   \n4           ZMYM1           79830     NaN     GRCh37          1   \n...           ...             ...     ...        ...        ...   \n73828   LINC00283       100874057     NaN     GRCh37         13   \n73829    C17orf50          146853     NaN     GRCh37         17   \n73830      MALRD1          340895     NaN     GRCh37         10   \n73831      MYO15B           80022     NaN     GRCh37         17   \n73832       MASP1            5648     NaN     GRCh37          3   \n\n       Start_Position  End_Position Strand              Consequence  \\\n0           114380884     114380884      +         missense_variant   \n1            21016724      21016724      +       synonymous_variant   \n2            11849446      11849446      +       synonymous_variant   \n3           158151458     158151458      +         missense_variant   \n4            35580705      35580705      +         missense_variant   \n...               ...           ...    ...                      ...   \n73828       103399819     103399819      +  downstream_gene_variant   \n73829        34095314      34095315      +  downstream_gene_variant   \n73830        19492731      19492731      +    upstream_gene_variant   \n73831        73609112      73609112      +  downstream_gene_variant   \n73832       186947661     186947661      +  downstream_gene_variant   \n\n      Variant_Classification  ... gencode_transcript_name  \\\n0          Missense_Mutation  ...              PTPN22-001   \n1                     Silent  ...               KIF17-009   \n2                     Silent  ...               MTHFR-001   \n3          Missense_Mutation  ...                CD1D-001   \n4          Missense_Mutation  ...               ZMYM1-001   \n...                      ...  ...                     ...   \n73828                3'Flank  ...                     NaN   \n73829                3'Flank  ...                     NaN   \n73830                5'Flank  ...                     NaN   \n73831                3'Flank  ...              MYO15B-001   \n73832                3'Flank  ...               MASP1-001   \n\n      gencode_transcript_status  \\\n0                         KNOWN   \n1                         KNOWN   \n2                         KNOWN   \n3                         KNOWN   \n4                         NOVEL   \n...                         ...   \n73828                       NaN   \n73829                       NaN   \n73830                       NaN   \n73831                     KNOWN   \n73832                     KNOWN   \n\n                             gencode_transcript_tags gencode_transcript_type  \\\n0                        basic|appris_principal|CCDS          protein_coding   \n1                basic|appris_candidate_longest|CCDS          protein_coding   \n2                        basic|appris_candidate|CCDS          protein_coding   \n3                        basic|appris_principal|CCDS          protein_coding   \n4      alternative_5_UTR|basic|appris_principal|CCDS          protein_coding   \n...                                              ...                     ...   \n73828                                            NaN                     NaN   \n73829                                            NaN                     NaN   \n73830                                            NaN                     NaN   \n73831                           sequence_error|basic    processed_transcript   \n73832                    basic|appris_principal|CCDS          protein_coding   \n\n      gene_id       gene_type     havana_transcript             ref_context  \\\n0         NaN  protein_coding  OTTHUMT00000033015.1   AGCTCCAGAAAGTCAAAAGAA   \n1         NaN  protein_coding  OTTHUMT00000276995.1   ACAGCCTGACGTCATATGAGT   \n2         NaN  protein_coding  OTTHUMT00000006538.1   AGGAAGCCGCCAGAGCACCGC   \n3         NaN  protein_coding  OTTHUMT00000058340.1   CGGGTTTATCGAAGCAGCTTC   \n4         NaN  protein_coding  OTTHUMT00000012705.1   TACCCTGCCTCGTCTTAAGAC   \n...       ...             ...                   ...                     ...   \n73828     0.0             NaN                   NaN   TTTCTGATAATTTTTTTTTAA   \n73829     0.0             NaN                   NaN  GCGCCTTCCTTGGGGGCTGTAG   \n73830     0.0             NaN                   NaN   TGACTGGATACGGAGCTCTCA   \n73831     NaN  protein_coding  OTTHUMT00000448172.2   GATCATGGGCGCATACCTGGT   \n73832     NaN  protein_coding  OTTHUMT00000344262.1   CATCAGCTTCCGGGAGAACTT   \n\n       secondary_variant_classification    transcript_id  \n0                                   NaN  ENST00000359785  \n1                                   NaN  ENST00000247986  \n2                                   NaN  ENST00000433342  \n3                                   NaN  ENST00000368171  \n4                                   NaN  ENST00000373330  \n...                                 ...              ...  \n73828                               NaN  ENST00000430111  \n73829                               NaN  ENST00000285023  \n73830                               NaN  ENST00000377266  \n73831                               NaN  ENST00000583560  \n73832                               NaN  ENST00000296280  \n\n[73833 rows x 298 columns]", "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Hugo_Symbol</th>\n      <th>Entrez_Gene_Id</th>\n      <th>Center</th>\n      <th>NCBI_Build</th>\n      <th>Chromosome</th>\n      <th>Start_Position</th>\n      <th>End_Position</th>\n      <th>Strand</th>\n      <th>Consequence</th>\n      <th>Variant_Classification</th>\n      <th>...</th>\n      <th>gencode_transcript_name</th>\n      <th>gencode_transcript_status</th>\n      <th>gencode_transcript_tags</th>\n      <th>gencode_transcript_type</th>\n      <th>gene_id</th>\n      <th>gene_type</th>\n      <th>havana_transcript</th>\n      <th>ref_context</th>\n      <th>secondary_variant_classification</th>\n      <th>transcript_id</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>PTPN22</td>\n      <td>26191</td>\n      <td>NaN</td>\n      <td>GRCh37</td>\n      <td>1</td>\n      <td>114380884</td>\n      <td>114380884</td>\n      <td>+</td>\n      <td>missense_variant</td>\n      <td>Missense_Mutation</td>\n      <td>...</td>\n      <td>PTPN22-001</td>\n      <td>KNOWN</td>\n      <td>basic|appris_principal|CCDS</td>\n      <td>protein_coding</td>\n      <td>NaN</td>\n      <td>protein_coding</td>\n      <td>OTTHUMT00000033015.1</td>\n      <td>AGCTCCAGAAAGTCAAAAGAA</td>\n      <td>NaN</td>\n      <td>ENST00000359785</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>KIF17</td>\n      <td>57576</td>\n      <td>NaN</td>\n      <td>GRCh37</td>\n      <td>1</td>\n      <td>21016724</td>\n      <td>21016724</td>\n      <td>+</td>\n      <td>synonymous_variant</td>\n      <td>Silent</td>\n      <td>...</td>\n      <td>KIF17-009</td>\n      <td>KNOWN</td>\n      <td>basic|appris_candidate_longest|CCDS</td>\n      <td>protein_coding</td>\n      <td>NaN</td>\n      <td>protein_coding</td>\n      <td>OTTHUMT00000276995.1</td>\n      <td>ACAGCCTGACGTCATATGAGT</td>\n      <td>NaN</td>\n      <td>ENST00000247986</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>C1orf167</td>\n      <td>284498</td>\n      <td>NaN</td>\n      <td>GRCh37</td>\n      <td>1</td>\n      <td>11849446</td>\n      <td>11849446</td>\n      <td>+</td>\n      <td>synonymous_variant</td>\n      <td>Silent</td>\n      <td>...</td>\n      <td>MTHFR-001</td>\n      <td>KNOWN</td>\n      <td>basic|appris_candidate|CCDS</td>\n      <td>protein_coding</td>\n      <td>NaN</td>\n      <td>protein_coding</td>\n      <td>OTTHUMT00000006538.1</td>\n      <td>AGGAAGCCGCCAGAGCACCGC</td>\n      <td>NaN</td>\n      <td>ENST00000433342</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>CD1D</td>\n      <td>912</td>\n      <td>NaN</td>\n      <td>GRCh37</td>\n      <td>1</td>\n      <td>158151458</td>\n      <td>158151458</td>\n      <td>+</td>\n      <td>missense_variant</td>\n      <td>Missense_Mutation</td>\n      <td>...</td>\n      <td>CD1D-001</td>\n      <td>KNOWN</td>\n      <td>basic|appris_principal|CCDS</td>\n      <td>protein_coding</td>\n      <td>NaN</td>\n      <td>protein_coding</td>\n      <td>OTTHUMT00000058340.1</td>\n      <td>CGGGTTTATCGAAGCAGCTTC</td>\n      <td>NaN</td>\n      <td>ENST00000368171</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>ZMYM1</td>\n      <td>79830</td>\n      <td>NaN</td>\n      <td>GRCh37</td>\n      <td>1</td>\n      <td>35580705</td>\n      <td>35580705</td>\n      <td>+</td>\n      <td>missense_variant</td>\n      <td>Missense_Mutation</td>\n      <td>...</td>\n      <td>ZMYM1-001</td>\n      <td>NOVEL</td>\n      <td>alternative_5_UTR|basic|appris_principal|CCDS</td>\n      <td>protein_coding</td>\n      <td>NaN</td>\n      <td>protein_coding</td>\n      <td>OTTHUMT00000012705.1</td>\n      <td>TACCCTGCCTCGTCTTAAGAC</td>\n      <td>NaN</td>\n      <td>ENST00000373330</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>73828</th>\n      <td>LINC00283</td>\n      <td>100874057</td>\n      <td>NaN</td>\n      <td>GRCh37</td>\n      <td>13</td>\n      <td>103399819</td>\n      <td>103399819</td>\n      <td>+</td>\n      <td>downstream_gene_variant</td>\n      <td>3'Flank</td>\n      <td>...</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>0.0</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>TTTCTGATAATTTTTTTTTAA</td>\n      <td>NaN</td>\n      <td>ENST00000430111</td>\n    </tr>\n    <tr>\n      <th>73829</th>\n      <td>C17orf50</td>\n      <td>146853</td>\n      <td>NaN</td>\n      <td>GRCh37</td>\n      <td>17</td>\n      <td>34095314</td>\n      <td>34095315</td>\n      <td>+</td>\n      <td>downstream_gene_variant</td>\n      <td>3'Flank</td>\n      <td>...</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>0.0</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>GCGCCTTCCTTGGGGGCTGTAG</td>\n      <td>NaN</td>\n      <td>ENST00000285023</td>\n    </tr>\n    <tr>\n      <th>73830</th>\n      <td>MALRD1</td>\n      <td>340895</td>\n      <td>NaN</td>\n      <td>GRCh37</td>\n      <td>10</td>\n      <td>19492731</td>\n      <td>19492731</td>\n      <td>+</td>\n      <td>upstream_gene_variant</td>\n      <td>5'Flank</td>\n      <td>...</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>0.0</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>TGACTGGATACGGAGCTCTCA</td>\n      <td>NaN</td>\n      <td>ENST00000377266</td>\n    </tr>\n    <tr>\n      <th>73831</th>\n      <td>MYO15B</td>\n      <td>80022</td>\n      <td>NaN</td>\n      <td>GRCh37</td>\n      <td>17</td>\n      <td>73609112</td>\n      <td>73609112</td>\n      <td>+</td>\n      <td>downstream_gene_variant</td>\n      <td>3'Flank</td>\n      <td>...</td>\n      <td>MYO15B-001</td>\n      <td>KNOWN</td>\n      <td>sequence_error|basic</td>\n      <td>processed_transcript</td>\n      <td>NaN</td>\n      <td>protein_coding</td>\n      <td>OTTHUMT00000448172.2</td>\n      <td>GATCATGGGCGCATACCTGGT</td>\n      <td>NaN</td>\n      <td>ENST00000583560</td>\n    </tr>\n    <tr>\n      <th>73832</th>\n      <td>MASP1</td>\n      <td>5648</td>\n      <td>NaN</td>\n      <td>GRCh37</td>\n      <td>3</td>\n      <td>186947661</td>\n      <td>186947661</td>\n      <td>+</td>\n      <td>downstream_gene_variant</td>\n      <td>3'Flank</td>\n      <td>...</td>\n      <td>MASP1-001</td>\n      <td>KNOWN</td>\n      <td>basic|appris_principal|CCDS</td>\n      <td>protein_coding</td>\n      <td>NaN</td>\n      <td>protein_coding</td>\n      <td>OTTHUMT00000344262.1</td>\n      <td>CATCAGCTTCCGGGAGAACTT</td>\n      <td>NaN</td>\n      <td>ENST00000296280</td>\n    </tr>\n  </tbody>\n</table>\n<p>73833 rows \u00d7 298 columns</p>\n</div>"}, "metadata": {}}]}, {"cell_type": "markdown", "source": "## Filter for mutations having ClinVar annotations", "metadata": {}}, {"cell_type": "code", "source": "mut_df = mut_df.loc[~mut_df['ClinVar_TYPE'].isna()]\nmut_df", "metadata": {"trusted": true, "tags": []}, "execution_count": 6, "outputs": [{"execution_count": 6, "output_type": "execute_result", "data": {"text/plain": "      Hugo_Symbol  Entrez_Gene_Id  Center NCBI_Build Chromosome  \\\n27           TP53            7157     NaN     GRCh37         17   \n65            APC             324     NaN     GRCh37          5   \n173          INSR            3643     NaN     GRCh37         19   \n248         TAAR9          134860     NaN     GRCh37          6   \n317          TP53            7157     NaN     GRCh37         17   \n...           ...             ...     ...        ...        ...   \n72809         APC             324     NaN     GRCh37          5   \n72952         APC             324     NaN     GRCh37          5   \n73011        PAX6            5080     NaN     GRCh37         11   \n73040        OCA2            4948     NaN     GRCh37         15   \n73482        FLT4            2324     NaN     GRCh37          5   \n\n       Start_Position  End_Position Strand  \\\n27            7577548       7577548      +   \n65          112175273     112175273      +   \n173           7122669       7122669      +   \n248         132859609     132859609      +   \n317           7577114       7577114      +   \n...               ...           ...    ...   \n72809       112174405     112174405      +   \n72952       112175727     112175736      +   \n73011        31812317      31812317      +   \n73040        28200305      28200305      +   \n73482       180043464     180043464      +   \n\n                                    Consequence Variant_Classification  ...  \\\n27                             missense_variant      Missense_Mutation  ...   \n65                                  stop_gained      Nonsense_Mutation  ...   \n173                            missense_variant      Missense_Mutation  ...   \n248                                   stop_lost       Nonstop_Mutation  ...   \n317                            missense_variant      Missense_Mutation  ...   \n...                                         ...                    ...  ...   \n72809                        frameshift_variant        Frame_Shift_Del  ...   \n72952                        frameshift_variant        Frame_Shift_Del  ...   \n73011                        frameshift_variant        Frame_Shift_Del  ...   \n73040  frameshift_variant,splice_region_variant        Frame_Shift_Del  ...   \n73482                          missense_variant      Missense_Mutation  ...   \n\n      gencode_transcript_name gencode_transcript_status  \\\n27                   TP53-001                     KNOWN   \n65                    APC-201                     KNOWN   \n173                  INSR-001                     KNOWN   \n248                 TAAR9-001                     KNOWN   \n317                  TP53-001                     KNOWN   \n...                       ...                       ...   \n72809                 APC-201                     KNOWN   \n72952                 APC-201                     KNOWN   \n73011                PAX6-008                     KNOWN   \n73040                OCA2-001                     KNOWN   \n73482                FLT4-001                     KNOWN   \n\n                                 gencode_transcript_tags  \\\n27                           basic|appris_principal|CCDS   \n65                           basic|appris_principal|CCDS   \n173                  basic|appris_candidate_longest|CCDS   \n248                         mRNA_end_NF|cds_end_NF|basic   \n317                          basic|appris_principal|CCDS   \n...                                                  ...   \n72809                        basic|appris_principal|CCDS   \n72952                        basic|appris_principal|CCDS   \n73011  not_organism_supported|basic|appris_candidate|...   \n73040                        basic|appris_principal|CCDS   \n73482                        basic|appris_principal|CCDS   \n\n      gencode_transcript_type gene_id               gene_type  \\\n27             protein_coding     NaN          protein_coding   \n65             protein_coding     NaN          protein_coding   \n173            protein_coding     NaN          protein_coding   \n248    polymorphic_pseudogene     NaN  polymorphic_pseudogene   \n317            protein_coding     NaN          protein_coding   \n...                       ...     ...                     ...   \n72809          protein_coding     NaN          protein_coding   \n72952          protein_coding     NaN          protein_coding   \n73011          protein_coding     NaN          protein_coding   \n73040          protein_coding     NaN          protein_coding   \n73482          protein_coding     NaN          protein_coding   \n\n          havana_transcript                     ref_context  \\\n27     OTTHUMT00000367397.1           CGGTTCATGCCGCCCATGCAG   \n65     OTTHUMT00000250738.2           AGCAGTGTCACAGCACCCTAG   \n173    OTTHUMT00000458544.1           GCAGTTTCTCGCTGCCAGGTC   \n248    OTTHUMT00000042254.2           CCTTCACTTCTAACAACTGCA   \n317    OTTHUMT00000367397.1           AGGACAGGCACAAACACGCAC   \n...                     ...                             ...   \n72809  OTTHUMT00000250738.2           AGTTGAACTCTGGAAGGCAAA   \n72952  OTTHUMT00000250738.2  GTTCAGAGGGTCCAGGTTCTTCCAGATGCT   \n73011  OTTHUMT00000099293.4           CTGCATATGTGGGGGGGTGTA   \n73040  OTTHUMT00000250823.1           GCTGGGTACCTTTTTTTGGAG   \n73482  OTTHUMT00000253527.4           CAGAATGTTCCGAGCAGCCAG   \n\n       secondary_variant_classification    transcript_id  \n27                                  NaN  ENST00000269305  \n65                                  NaN  ENST00000257430  \n173                                 NaN  ENST00000302850  \n248                                 NaN  ENST00000434551  \n317                                 NaN  ENST00000269305  \n...                                 ...              ...  \n72809                               NaN  ENST00000257430  \n72952                               NaN  ENST00000257430  \n73011                               NaN  ENST00000419022  \n73040                   Frame_Shift_Del  ENST00000354638  \n73482                               NaN  ENST00000261937  \n\n[475 rows x 298 columns]", "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Hugo_Symbol</th>\n      <th>Entrez_Gene_Id</th>\n      <th>Center</th>\n      <th>NCBI_Build</th>\n      <th>Chromosome</th>\n      <th>Start_Position</th>\n      <th>End_Position</th>\n      <th>Strand</th>\n      <th>Consequence</th>\n      <th>Variant_Classification</th>\n      <th>...</th>\n      <th>gencode_transcript_name</th>\n      <th>gencode_transcript_status</th>\n      <th>gencode_transcript_tags</th>\n      <th>gencode_transcript_type</th>\n      <th>gene_id</th>\n      <th>gene_type</th>\n      <th>havana_transcript</th>\n      <th>ref_context</th>\n      <th>secondary_variant_classification</th>\n      <th>transcript_id</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>27</th>\n      <td>TP53</td>\n      <td>7157</td>\n      <td>NaN</td>\n      <td>GRCh37</td>\n      <td>17</td>\n      <td>7577548</td>\n      <td>7577548</td>\n      <td>+</td>\n      <td>missense_variant</td>\n      <td>Missense_Mutation</td>\n      <td>...</td>\n      <td>TP53-001</td>\n      <td>KNOWN</td>\n      <td>basic|appris_principal|CCDS</td>\n      <td>protein_coding</td>\n      <td>NaN</td>\n      <td>protein_coding</td>\n      <td>OTTHUMT00000367397.1</td>\n      <td>CGGTTCATGCCGCCCATGCAG</td>\n      <td>NaN</td>\n      <td>ENST00000269305</td>\n    </tr>\n    <tr>\n      <th>65</th>\n      <td>APC</td>\n      <td>324</td>\n      <td>NaN</td>\n      <td>GRCh37</td>\n      <td>5</td>\n      <td>112175273</td>\n      <td>112175273</td>\n      <td>+</td>\n      <td>stop_gained</td>\n      <td>Nonsense_Mutation</td>\n      <td>...</td>\n      <td>APC-201</td>\n      <td>KNOWN</td>\n      <td>basic|appris_principal|CCDS</td>\n      <td>protein_coding</td>\n      <td>NaN</td>\n      <td>protein_coding</td>\n      <td>OTTHUMT00000250738.2</td>\n      <td>AGCAGTGTCACAGCACCCTAG</td>\n      <td>NaN</td>\n      <td>ENST00000257430</td>\n    </tr>\n    <tr>\n      <th>173</th>\n      <td>INSR</td>\n      <td>3643</td>\n      <td>NaN</td>\n      <td>GRCh37</td>\n      <td>19</td>\n      <td>7122669</td>\n      <td>7122669</td>\n      <td>+</td>\n      <td>missense_variant</td>\n      <td>Missense_Mutation</td>\n      <td>...</td>\n      <td>INSR-001</td>\n      <td>KNOWN</td>\n      <td>basic|appris_candidate_longest|CCDS</td>\n      <td>protein_coding</td>\n      <td>NaN</td>\n      <td>protein_coding</td>\n      <td>OTTHUMT00000458544.1</td>\n      <td>GCAGTTTCTCGCTGCCAGGTC</td>\n      <td>NaN</td>\n      <td>ENST00000302850</td>\n    </tr>\n    <tr>\n      <th>248</th>\n      <td>TAAR9</td>\n      <td>134860</td>\n      <td>NaN</td>\n      <td>GRCh37</td>\n      <td>6</td>\n      <td>132859609</td>\n      <td>132859609</td>\n      <td>+</td>\n      <td>stop_lost</td>\n      <td>Nonstop_Mutation</td>\n      <td>...</td>\n      <td>TAAR9-001</td>\n      <td>KNOWN</td>\n      <td>mRNA_end_NF|cds_end_NF|basic</td>\n      <td>polymorphic_pseudogene</td>\n      <td>NaN</td>\n      <td>polymorphic_pseudogene</td>\n      <td>OTTHUMT00000042254.2</td>\n      <td>CCTTCACTTCTAACAACTGCA</td>\n      <td>NaN</td>\n      <td>ENST00000434551</td>\n    </tr>\n    <tr>\n      <th>317</th>\n      <td>TP53</td>\n      <td>7157</td>\n      <td>NaN</td>\n      <td>GRCh37</td>\n      <td>17</td>\n      <td>7577114</td>\n      <td>7577114</td>\n      <td>+</td>\n      <td>missense_variant</td>\n      <td>Missense_Mutation</td>\n      <td>...</td>\n      <td>TP53-001</td>\n      <td>KNOWN</td>\n      <td>basic|appris_principal|CCDS</td>\n      <td>protein_coding</td>\n      <td>NaN</td>\n      <td>protein_coding</td>\n      <td>OTTHUMT00000367397.1</td>\n      <td>AGGACAGGCACAAACACGCAC</td>\n      <td>NaN</td>\n      <td>ENST00000269305</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>72809</th>\n      <td>APC</td>\n      <td>324</td>\n      <td>NaN</td>\n      <td>GRCh37</td>\n      <td>5</td>\n      <td>112174405</td>\n      <td>112174405</td>\n      <td>+</td>\n      <td>frameshift_variant</td>\n      <td>Frame_Shift_Del</td>\n      <td>...</td>\n      <td>APC-201</td>\n      <td>KNOWN</td>\n      <td>basic|appris_principal|CCDS</td>\n      <td>protein_coding</td>\n      <td>NaN</td>\n      <td>protein_coding</td>\n      <td>OTTHUMT00000250738.2</td>\n      <td>AGTTGAACTCTGGAAGGCAAA</td>\n      <td>NaN</td>\n      <td>ENST00000257430</td>\n    </tr>\n    <tr>\n      <th>72952</th>\n      <td>APC</td>\n      <td>324</td>\n      <td>NaN</td>\n      <td>GRCh37</td>\n      <td>5</td>\n      <td>112175727</td>\n      <td>112175736</td>\n      <td>+</td>\n      <td>frameshift_variant</td>\n      <td>Frame_Shift_Del</td>\n      <td>...</td>\n      <td>APC-201</td>\n      <td>KNOWN</td>\n      <td>basic|appris_principal|CCDS</td>\n      <td>protein_coding</td>\n      <td>NaN</td>\n      <td>protein_coding</td>\n      <td>OTTHUMT00000250738.2</td>\n      <td>GTTCAGAGGGTCCAGGTTCTTCCAGATGCT</td>\n      <td>NaN</td>\n      <td>ENST00000257430</td>\n    </tr>\n    <tr>\n      <th>73011</th>\n      <td>PAX6</td>\n      <td>5080</td>\n      <td>NaN</td>\n      <td>GRCh37</td>\n      <td>11</td>\n      <td>31812317</td>\n      <td>31812317</td>\n      <td>+</td>\n      <td>frameshift_variant</td>\n      <td>Frame_Shift_Del</td>\n      <td>...</td>\n      <td>PAX6-008</td>\n      <td>KNOWN</td>\n      <td>not_organism_supported|basic|appris_candidate|...</td>\n      <td>protein_coding</td>\n      <td>NaN</td>\n      <td>protein_coding</td>\n      <td>OTTHUMT00000099293.4</td>\n      <td>CTGCATATGTGGGGGGGTGTA</td>\n      <td>NaN</td>\n      <td>ENST00000419022</td>\n    </tr>\n    <tr>\n      <th>73040</th>\n      <td>OCA2</td>\n      <td>4948</td>\n      <td>NaN</td>\n      <td>GRCh37</td>\n      <td>15</td>\n      <td>28200305</td>\n      <td>28200305</td>\n      <td>+</td>\n      <td>frameshift_variant,splice_region_variant</td>\n      <td>Frame_Shift_Del</td>\n      <td>...</td>\n      <td>OCA2-001</td>\n      <td>KNOWN</td>\n      <td>basic|appris_principal|CCDS</td>\n      <td>protein_coding</td>\n      <td>NaN</td>\n      <td>protein_coding</td>\n      <td>OTTHUMT00000250823.1</td>\n      <td>GCTGGGTACCTTTTTTTGGAG</td>\n      <td>Frame_Shift_Del</td>\n      <td>ENST00000354638</td>\n    </tr>\n    <tr>\n      <th>73482</th>\n      <td>FLT4</td>\n      <td>2324</td>\n      <td>NaN</td>\n      <td>GRCh37</td>\n      <td>5</td>\n      <td>180043464</td>\n      <td>180043464</td>\n      <td>+</td>\n      <td>missense_variant</td>\n      <td>Missense_Mutation</td>\n      <td>...</td>\n      <td>FLT4-001</td>\n      <td>KNOWN</td>\n      <td>basic|appris_principal|CCDS</td>\n      <td>protein_coding</td>\n      <td>NaN</td>\n      <td>protein_coding</td>\n      <td>OTTHUMT00000253527.4</td>\n      <td>CAGAATGTTCCGAGCAGCCAG</td>\n      <td>NaN</td>\n      <td>ENST00000261937</td>\n    </tr>\n  </tbody>\n</table>\n<p>475 rows \u00d7 298 columns</p>\n</div>"}, "metadata": {}}]}, {"cell_type": "code", "source": "len(mut_df.loc[~mut_df['ClinVar_TYPE'].isna()]['Tumor_Sample_Barcode'].unique())", "metadata": {"trusted": true, "tags": []}, "execution_count": 7, "outputs": [{"execution_count": 7, "output_type": "execute_result", "data": {"text/plain": "93"}, "metadata": {}}]}, {"cell_type": "markdown", "source": "## Read TCGA clinical sample file", "metadata": {}}, {"cell_type": "code", "source": "clinical_df = pd.read_csv(\"/mnt/project/data/coad_cptac_2019/data_clinical_sample.txt\",sep=\"\\t\",skiprows=4)\nclinical_df", "metadata": {"trusted": true, "tags": []}, "execution_count": 8, "outputs": [{"execution_count": 8, "output_type": "execute_result", "data": {"text/plain": "    PATIENT_ID SAMPLE_ID SPECIMEN_PRESERVATION  SEQUENCED  COPY_NUMBER  \\\n0      01CO001   01CO001         Frozen Tissue          1            1   \n1      01CO005   01CO005         Frozen Tissue          1            1   \n2      01CO006   01CO006         Frozen Tissue          1            1   \n3      01CO008   01CO008         Frozen Tissue          1            1   \n4      01CO013   01CO013         Frozen Tissue          1            1   \n..         ...       ...                   ...        ...          ...   \n105    05CO014   05CO014       [Not Available]          1            1   \n106    05CO055   05CO055       [Not Available]          1            1   \n107    11CO059   11CO059       [Not Available]          1            1   \n108    16CO012   16CO012       [Not Available]          1            1   \n109    24CO005   24CO005       [Not Available]          1            1   \n\n     MRNA_DATA  MICRORNA_DATA  METHYLATION_STATUS  PROTEIN  PHOSPHOPROTEIN  \\\n0            1              1                   0        0               0   \n1            1              1                   0        1               1   \n2            1              1                   0        1               1   \n3            1              1                   0        1               1   \n4            1              1                   0        1               1   \n..         ...            ...                 ...      ...             ...   \n105          1              1                   0        0               0   \n106          1              1                   0        0               0   \n107          1              1                   0        0               0   \n108          1              1                   0        0               0   \n109          1              1                   0        0               0   \n\n    MSI_STATUS PATHOLOGY_STATUS      PRIMARY_SITE ONCOTREE_CODE  \\\n0          MSS        Malignant     Sigmoid Colon          COAD   \n1          MSS        Malignant     Sigmoid Colon          COAD   \n2          MSS        Malignant   Ascending Colon          COAD   \n3          MSS        Malignant  Descending Colon          COAD   \n4          MSS        Malignant     Sigmoid Colon          COAD   \n..         ...              ...               ...           ...   \n105        MSS  [Not Available]   [Not Available]          COAD   \n106        MSS  [Not Available]     Sigmoid Colon          COAD   \n107      MSI-H  [Not Available]   Ascending Colon          COAD   \n108        MSS  [Not Available]     Sigmoid Colon          COAD   \n109        MSS  [Not Available]     Sigmoid Colon          COAD   \n\n           CANCER_TYPE  CANCER_TYPE_DETAILED SOMATIC_STATUS  TMB_NONSYNONYMOUS  \n0    Colorectal Cancer  Colon Adenocarcinoma        Matched           2.566667  \n1    Colorectal Cancer  Colon Adenocarcinoma        Matched           4.366667  \n2    Colorectal Cancer  Colon Adenocarcinoma        Matched           2.700000  \n3    Colorectal Cancer  Colon Adenocarcinoma        Matched           5.166667  \n4    Colorectal Cancer  Colon Adenocarcinoma        Matched           2.633333  \n..                 ...                   ...            ...                ...  \n105  Colorectal Cancer  Colon Adenocarcinoma        Matched           2.866667  \n106  Colorectal Cancer  Colon Adenocarcinoma        Matched           3.600000  \n107  Colorectal Cancer  Colon Adenocarcinoma        Matched         102.033333  \n108  Colorectal Cancer  Colon Adenocarcinoma        Matched           3.733333  \n109  Colorectal Cancer  Colon Adenocarcinoma        Matched           3.200000  \n\n[110 rows x 18 columns]", "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>PATIENT_ID</th>\n      <th>SAMPLE_ID</th>\n      <th>SPECIMEN_PRESERVATION</th>\n      <th>SEQUENCED</th>\n      <th>COPY_NUMBER</th>\n      <th>MRNA_DATA</th>\n      <th>MICRORNA_DATA</th>\n      <th>METHYLATION_STATUS</th>\n      <th>PROTEIN</th>\n      <th>PHOSPHOPROTEIN</th>\n      <th>MSI_STATUS</th>\n      <th>PATHOLOGY_STATUS</th>\n      <th>PRIMARY_SITE</th>\n      <th>ONCOTREE_CODE</th>\n      <th>CANCER_TYPE</th>\n      <th>CANCER_TYPE_DETAILED</th>\n      <th>SOMATIC_STATUS</th>\n      <th>TMB_NONSYNONYMOUS</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>01CO001</td>\n      <td>01CO001</td>\n      <td>Frozen Tissue</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n      <td>MSS</td>\n      <td>Malignant</td>\n      <td>Sigmoid Colon</td>\n      <td>COAD</td>\n      <td>Colorectal Cancer</td>\n      <td>Colon Adenocarcinoma</td>\n      <td>Matched</td>\n      <td>2.566667</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>01CO005</td>\n      <td>01CO005</td>\n      <td>Frozen Tissue</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>0</td>\n      <td>1</td>\n      <td>1</td>\n      <td>MSS</td>\n      <td>Malignant</td>\n      <td>Sigmoid Colon</td>\n      <td>COAD</td>\n      <td>Colorectal Cancer</td>\n      <td>Colon Adenocarcinoma</td>\n      <td>Matched</td>\n      <td>4.366667</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>01CO006</td>\n      <td>01CO006</td>\n      <td>Frozen Tissue</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>0</td>\n      <td>1</td>\n      <td>1</td>\n      <td>MSS</td>\n      <td>Malignant</td>\n      <td>Ascending Colon</td>\n      <td>COAD</td>\n      <td>Colorectal Cancer</td>\n      <td>Colon Adenocarcinoma</td>\n      <td>Matched</td>\n      <td>2.700000</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>01CO008</td>\n      <td>01CO008</td>\n      <td>Frozen Tissue</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>0</td>\n      <td>1</td>\n      <td>1</td>\n      <td>MSS</td>\n      <td>Malignant</td>\n      <td>Descending Colon</td>\n      <td>COAD</td>\n      <td>Colorectal Cancer</td>\n      <td>Colon Adenocarcinoma</td>\n      <td>Matched</td>\n      <td>5.166667</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>01CO013</td>\n      <td>01CO013</td>\n      <td>Frozen Tissue</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>0</td>\n      <td>1</td>\n      <td>1</td>\n      <td>MSS</td>\n      <td>Malignant</td>\n      <td>Sigmoid Colon</td>\n      <td>COAD</td>\n      <td>Colorectal Cancer</td>\n      <td>Colon Adenocarcinoma</td>\n      <td>Matched</td>\n      <td>2.633333</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>105</th>\n      <td>05CO014</td>\n      <td>05CO014</td>\n      <td>[Not Available]</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n      <td>MSS</td>\n      <td>[Not Available]</td>\n      <td>[Not Available]</td>\n      <td>COAD</td>\n      <td>Colorectal Cancer</td>\n      <td>Colon Adenocarcinoma</td>\n      <td>Matched</td>\n      <td>2.866667</td>\n    </tr>\n    <tr>\n      <th>106</th>\n      <td>05CO055</td>\n      <td>05CO055</td>\n      <td>[Not Available]</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n      <td>MSS</td>\n      <td>[Not Available]</td>\n      <td>Sigmoid Colon</td>\n      <td>COAD</td>\n      <td>Colorectal Cancer</td>\n      <td>Colon Adenocarcinoma</td>\n      <td>Matched</td>\n      <td>3.600000</td>\n    </tr>\n    <tr>\n      <th>107</th>\n      <td>11CO059</td>\n      <td>11CO059</td>\n      <td>[Not Available]</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n      <td>MSI-H</td>\n      <td>[Not Available]</td>\n      <td>Ascending Colon</td>\n      <td>COAD</td>\n      <td>Colorectal Cancer</td>\n      <td>Colon Adenocarcinoma</td>\n      <td>Matched</td>\n      <td>102.033333</td>\n    </tr>\n    <tr>\n      <th>108</th>\n      <td>16CO012</td>\n      <td>16CO012</td>\n      <td>[Not Available]</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n      <td>MSS</td>\n      <td>[Not Available]</td>\n      <td>Sigmoid Colon</td>\n      <td>COAD</td>\n      <td>Colorectal Cancer</td>\n      <td>Colon Adenocarcinoma</td>\n      <td>Matched</td>\n      <td>3.733333</td>\n    </tr>\n    <tr>\n      <th>109</th>\n      <td>24CO005</td>\n      <td>24CO005</td>\n      <td>[Not Available]</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n      <td>MSS</td>\n      <td>[Not Available]</td>\n      <td>Sigmoid Colon</td>\n      <td>COAD</td>\n      <td>Colorectal Cancer</td>\n      <td>Colon Adenocarcinoma</td>\n      <td>Matched</td>\n      <td>3.200000</td>\n    </tr>\n  </tbody>\n</table>\n<p>110 rows \u00d7 18 columns</p>\n</div>"}, "metadata": {}}]}, {"cell_type": "markdown", "source": "## Filter for patients having clinvar annotated mutations", "metadata": {}}, {"cell_type": "code", "source": "clinical_df.loc[clinical_df['SAMPLE_ID'].isin(mut_df.loc[~mut_df['ClinVar_TYPE'].isna()]['Tumor_Sample_Barcode'].unique())]", "metadata": {"trusted": true, "tags": []}, "execution_count": 9, "outputs": [{"execution_count": 9, "output_type": "execute_result", "data": {"text/plain": "    PATIENT_ID SAMPLE_ID SPECIMEN_PRESERVATION  SEQUENCED  COPY_NUMBER  \\\n0      01CO001   01CO001         Frozen Tissue          1            1   \n1      01CO005   01CO005         Frozen Tissue          1            1   \n3      01CO008   01CO008         Frozen Tissue          1            1   \n4      01CO013   01CO013         Frozen Tissue          1            1   \n5      01CO014   01CO014         Frozen Tissue          1            1   \n..         ...       ...                   ...        ...          ...   \n102    22CO006   22CO006         Frozen Tissue          1            1   \n104    05CO005   05CO005       [Not Available]          1            1   \n105    05CO014   05CO014       [Not Available]          1            1   \n107    11CO059   11CO059       [Not Available]          1            1   \n109    24CO005   24CO005       [Not Available]          1            1   \n\n     MRNA_DATA  MICRORNA_DATA  METHYLATION_STATUS  PROTEIN  PHOSPHOPROTEIN  \\\n0            1              1                   0        0               0   \n1            1              1                   0        1               1   \n3            1              1                   0        1               1   \n4            1              1                   0        1               1   \n5            1              1                   0        1               1   \n..         ...            ...                 ...      ...             ...   \n102          1              1                   0        1               1   \n104          1              1                   0        0               0   \n105          1              1                   0        0               0   \n107          1              1                   0        0               0   \n109          1              1                   0        0               0   \n\n    MSI_STATUS PATHOLOGY_STATUS      PRIMARY_SITE ONCOTREE_CODE  \\\n0          MSS        Malignant     Sigmoid Colon          COAD   \n1          MSS        Malignant     Sigmoid Colon          COAD   \n3          MSS        Malignant  Descending Colon          COAD   \n4          MSS        Malignant     Sigmoid Colon          COAD   \n5        MSI-H        Malignant   Hepatix Flexure          COAD   \n..         ...              ...               ...           ...   \n102      MSI-H        Malignant   Ascending Colon          COAD   \n104        MSS  [Not Available]     Sigmoid Colon          COAD   \n105        MSS  [Not Available]   [Not Available]          COAD   \n107      MSI-H  [Not Available]   Ascending Colon          COAD   \n109        MSS  [Not Available]     Sigmoid Colon          COAD   \n\n           CANCER_TYPE  CANCER_TYPE_DETAILED SOMATIC_STATUS  TMB_NONSYNONYMOUS  \n0    Colorectal Cancer  Colon Adenocarcinoma        Matched           2.566667  \n1    Colorectal Cancer  Colon Adenocarcinoma        Matched           4.366667  \n3    Colorectal Cancer  Colon Adenocarcinoma        Matched           5.166667  \n4    Colorectal Cancer  Colon Adenocarcinoma        Matched           2.633333  \n5    Colorectal Cancer  Colon Adenocarcinoma        Matched          33.900000  \n..                 ...                   ...            ...                ...  \n102  Colorectal Cancer  Colon Adenocarcinoma        Matched          34.933333  \n104  Colorectal Cancer  Colon Adenocarcinoma        Matched           4.466667  \n105  Colorectal Cancer  Colon Adenocarcinoma        Matched           2.866667  \n107  Colorectal Cancer  Colon Adenocarcinoma        Matched         102.033333  \n109  Colorectal Cancer  Colon Adenocarcinoma        Matched           3.200000  \n\n[93 rows x 18 columns]", "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>PATIENT_ID</th>\n      <th>SAMPLE_ID</th>\n      <th>SPECIMEN_PRESERVATION</th>\n      <th>SEQUENCED</th>\n      <th>COPY_NUMBER</th>\n      <th>MRNA_DATA</th>\n      <th>MICRORNA_DATA</th>\n      <th>METHYLATION_STATUS</th>\n      <th>PROTEIN</th>\n      <th>PHOSPHOPROTEIN</th>\n      <th>MSI_STATUS</th>\n      <th>PATHOLOGY_STATUS</th>\n      <th>PRIMARY_SITE</th>\n      <th>ONCOTREE_CODE</th>\n      <th>CANCER_TYPE</th>\n      <th>CANCER_TYPE_DETAILED</th>\n      <th>SOMATIC_STATUS</th>\n      <th>TMB_NONSYNONYMOUS</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>01CO001</td>\n      <td>01CO001</td>\n      <td>Frozen Tissue</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n      <td>MSS</td>\n      <td>Malignant</td>\n      <td>Sigmoid Colon</td>\n      <td>COAD</td>\n      <td>Colorectal Cancer</td>\n      <td>Colon Adenocarcinoma</td>\n      <td>Matched</td>\n      <td>2.566667</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>01CO005</td>\n      <td>01CO005</td>\n      <td>Frozen Tissue</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>0</td>\n      <td>1</td>\n      <td>1</td>\n      <td>MSS</td>\n      <td>Malignant</td>\n      <td>Sigmoid Colon</td>\n      <td>COAD</td>\n      <td>Colorectal Cancer</td>\n      <td>Colon Adenocarcinoma</td>\n      <td>Matched</td>\n      <td>4.366667</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>01CO008</td>\n      <td>01CO008</td>\n      <td>Frozen Tissue</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>0</td>\n      <td>1</td>\n      <td>1</td>\n      <td>MSS</td>\n      <td>Malignant</td>\n      <td>Descending Colon</td>\n      <td>COAD</td>\n      <td>Colorectal Cancer</td>\n      <td>Colon Adenocarcinoma</td>\n      <td>Matched</td>\n      <td>5.166667</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>01CO013</td>\n      <td>01CO013</td>\n      <td>Frozen Tissue</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>0</td>\n      <td>1</td>\n      <td>1</td>\n      <td>MSS</td>\n      <td>Malignant</td>\n      <td>Sigmoid Colon</td>\n      <td>COAD</td>\n      <td>Colorectal Cancer</td>\n      <td>Colon Adenocarcinoma</td>\n      <td>Matched</td>\n      <td>2.633333</td>\n    </tr>\n    <tr>\n      <th>5</th>\n      <td>01CO014</td>\n      <td>01CO014</td>\n      <td>Frozen Tissue</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>0</td>\n      <td>1</td>\n      <td>1</td>\n      <td>MSI-H</td>\n      <td>Malignant</td>\n      <td>Hepatix Flexure</td>\n      <td>COAD</td>\n      <td>Colorectal Cancer</td>\n      <td>Colon Adenocarcinoma</td>\n      <td>Matched</td>\n      <td>33.900000</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>102</th>\n      <td>22CO006</td>\n      <td>22CO006</td>\n      <td>Frozen Tissue</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>0</td>\n      <td>1</td>\n      <td>1</td>\n      <td>MSI-H</td>\n      <td>Malignant</td>\n      <td>Ascending Colon</td>\n      <td>COAD</td>\n      <td>Colorectal Cancer</td>\n      <td>Colon Adenocarcinoma</td>\n      <td>Matched</td>\n      <td>34.933333</td>\n    </tr>\n    <tr>\n      <th>104</th>\n      <td>05CO005</td>\n      <td>05CO005</td>\n      <td>[Not Available]</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n      <td>MSS</td>\n      <td>[Not Available]</td>\n      <td>Sigmoid Colon</td>\n      <td>COAD</td>\n      <td>Colorectal Cancer</td>\n      <td>Colon Adenocarcinoma</td>\n      <td>Matched</td>\n      <td>4.466667</td>\n    </tr>\n    <tr>\n      <th>105</th>\n      <td>05CO014</td>\n      <td>05CO014</td>\n      <td>[Not Available]</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n      <td>MSS</td>\n      <td>[Not Available]</td>\n      <td>[Not Available]</td>\n      <td>COAD</td>\n      <td>Colorectal Cancer</td>\n      <td>Colon Adenocarcinoma</td>\n      <td>Matched</td>\n      <td>2.866667</td>\n    </tr>\n    <tr>\n      <th>107</th>\n      <td>11CO059</td>\n      <td>11CO059</td>\n      <td>[Not Available]</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n      <td>MSI-H</td>\n      <td>[Not Available]</td>\n      <td>Ascending Colon</td>\n      <td>COAD</td>\n      <td>Colorectal Cancer</td>\n      <td>Colon Adenocarcinoma</td>\n      <td>Matched</td>\n      <td>102.033333</td>\n    </tr>\n    <tr>\n      <th>109</th>\n      <td>24CO005</td>\n      <td>24CO005</td>\n      <td>[Not Available]</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n      <td>MSS</td>\n      <td>[Not Available]</td>\n      <td>Sigmoid Colon</td>\n      <td>COAD</td>\n      <td>Colorectal Cancer</td>\n      <td>Colon Adenocarcinoma</td>\n      <td>Matched</td>\n      <td>3.200000</td>\n    </tr>\n  </tbody>\n</table>\n<p>93 rows \u00d7 18 columns</p>\n</div>"}, "metadata": {}}]}, {"cell_type": "markdown", "source": "## Make mapping between cancer type and Mondo ID", "metadata": {}}, {"cell_type": "code", "source": "cancer_mondo={}\nfor c in clinical_df['CANCER_TYPE_DETAILED'].unique():\n    mondoid=query_mondo(c)\n    cancer_mondo[c]=mondoid", "metadata": {"trusted": true, "tags": []}, "execution_count": 10, "outputs": []}, {"cell_type": "code", "source": "print(cancer_mondo)", "metadata": {"trusted": true, "tags": []}, "execution_count": 11, "outputs": [{"name": "stdout", "text": "{'Colon Adenocarcinoma': 'http://purl.obolibrary.org/obo/MONDO_0001552'}\n", "output_type": "stream"}]}, {"cell_type": "markdown", "source": "## Load Mondo ontology", "metadata": {}}, {"cell_type": "code", "source": "mondo_json = requests.get('https://github.com/monarch-initiative/mondo/releases/latest/download/mondo.json').json()\nmondo_nodes = mondo_json['graphs'][0]['nodes']\nmondo_nodes[0]", "metadata": {"trusted": true, "tags": []}, "execution_count": 12, "outputs": [{"execution_count": 12, "output_type": "execute_result", "data": {"text/plain": "{'id': 'http://identifiers.org/hgnc/10001', 'lbl': 'RGS5', 'type': 'CLASS'}"}, "metadata": {}}]}, {"cell_type": "code", "source": "def get_gene_mondo_id(target_lbl=None):\n    #target_lbl = 'SHH'\n    matching_node = None\n\n    for node in mondo_json['graphs'][0]['nodes']:\n        if 'lbl' in node and node['lbl'] == target_lbl:\n            matching_node = node\n            break\n\n    if matching_node:\n        #print(f\"Matching node found:{matching_node['id']}\")\n        return matching_node['id']\n    else:\n        pass\n        #print(f\"No node with lbl '{target_lbl}' found.\")", "metadata": {"trusted": true, "tags": []}, "execution_count": 13, "outputs": []}, {"cell_type": "code", "source": "get_gene_mondo_id('SHH')", "metadata": {"trusted": true, "tags": []}, "execution_count": 14, "outputs": [{"execution_count": 14, "output_type": "execute_result", "data": {"text/plain": "'http://identifiers.org/hgnc/10848'"}, "metadata": {}}]}, {"cell_type": "markdown", "source": "## Add nodes to a graph", "metadata": {}}, {"cell_type": "code", "source": "g = Graph()\n\n# Define namespaces\ntt = Namespace(\"http://tinytcga.org/\")  # tiny tcga namespace\nmondo = Namespace(\"http://purl.obolibrary.org/obo/mondo#\")  # Mondo namespace\n\n# Define properties\nhasHugoSymbol = tt.hasHugoSymbol\nisCancerTypeOf = tt.isCancerTypeOf\n\n# Add data to the graph\nfor index, row in clinical_df.loc[clinical_df['SAMPLE_ID'].isin(mut_df.loc[~mut_df['ClinVar_TYPE'].isna()]['Tumor_Sample_Barcode'].unique())].iterrows():\n    #print(f\"{row['CANCER_TYPE_DETAILED']} {row['SAMPLE_ID']}\")\n    sample = f\"http://tinytcga.org/{row['SAMPLE_ID']}\"\n\n    #sample = f\"tt.{row['SAMPLE_ID']}\"\n    mondoid=cancer_mondo.get(row['CANCER_TYPE_DETAILED'])\n    #print(mondoid)\n    #g.add(sample, isCancerTypeOf, URIRef(mondoid))\n    g.add((URIRef(sample), isCancerTypeOf, URIRef(mondoid)))\n    for j, mrow in mut_df[mut_df['Tumor_Sample_Barcode']==row['SAMPLE_ID']].iterrows():\n        #print(mrow['Hugo_Symbol'])\n        gene_mondo=get_gene_mondo_id(mrow['Hugo_Symbol'])\n        if gene_mondo is not None:\n            g.add((URIRef(sample), hasHugoSymbol, URIRef(gene_mondo)))\n", "metadata": {"trusted": true, "tags": []}, "execution_count": 15, "outputs": []}, {"cell_type": "markdown", "source": "## Serialize the graph to a ttl", "metadata": {}}, {"cell_type": "code", "source": "g.serialize(\"tinytcga.ttl\")", "metadata": {"trusted": true, "tags": []}, "execution_count": 16, "outputs": [{"execution_count": 16, "output_type": "execute_result", "data": {"text/plain": "<Graph identifier=Nd31bd351b075470382740518518f6cf4 (<class 'rdflib.graph.Graph'>)>"}, "metadata": {}}]}, {"cell_type": "code", "source": "", "metadata": {}, "execution_count": null, "outputs": []}]}